### ISSUE TITLE
Improve caching documentation

### ISSUE THREAD

**Issue by keflavich** [astropy/astroquery#1239]
There's very little documentation on the caching process, and users sometimes complain about the cache filling up. 

In short, we need documentation stating:

1) The default location is `~/.astropy/cache/astroquery`
2) [get_cache_dir](http://docs.astropy.org/en/stable/api/astropy.config.get_cache_dir.html#astropy.config.get_cache_dir) can be used to find the cache dir
3) some documentation about how [clear_download_cache](http://docs.astropy.org/en/stable/api/astropy.utils.data.clear_download_cache.html) can be used to clear the cache, with examples

**Comment by keflavich** [comment]
Dangit, I searched for 'cache' but not 'caching'.  My memory fails.

**Comment by bsipocz** [comment]
I only remembered because of the recent simbad issues 😄

**Comment by Himanshu-Garg** [comment]
Anyone on this?

**Comment by bsipocz** [comment]
no one, please go ahead!

**Comment by Himanshu-Garg** [comment]
Can you pls mention the doc where i have to insert the new info

I cannot find it in docs folder

**Comment by Himanshu-Garg** [comment]
here cache folder is not showing in the cloned repo because the repo cloned is of older version but the following page is available in newer version-
http://docs.astropy.org/en/stable/api/astropy.config.get_cache_dir.html#astropy.config.get_cache_dir

so, how to make changes in this page?

**Comment by UdokaVrede** [comment]
can I take this up

**Comment by bsipocz** [comment]
I'm afraid this is a bit miscategorized to be package-novice in a sense it is best suited for users running into cache issues. E.g. we cannot give much guidance on where to document these things to be the most effective, where the users would most likely see it.
Maybe a frequent solutions to issues on the opening page,  Troubleshouting? 

So you certainly have a go with it, but it would be the most impactful if there was input from users. You can reach some of them on the astroquery slack channel, or the astropy mailing list, or ping the people in issues from the past 1-2 years where the solution was something cache related where the extra docs would help.

So, in summary, you've welcome to work on this, but it is not a straightforward issue because of the meta layer.

**Comment by UdokaVrede** [comment]
I DID NOT KNOW IT WAS THIS VAST, I WOULD SKIP THIS WHILE I GET FAMILIAR
WITH ASTROPY. I WOULD APPRECIATE LINKS TO package-novice issues.Thanks

**Comment by bsipocz** [comment]
@ceb8 - while the cache stuff is still fresh, could you have a  look at the checklist above and see what else is missing in the updated docs?

**Comment by ceb8** [comment]
@keflavich Can you explain what you mean by the last item?

**Comment by keflavich** [comment]
My idea was that we should have global documentation about the cache & timeout parameters, and each individual function's docstring that describes those parameters briefly should link back to the (longer and more complete) global documentation.

**Comment by JohannesBuchner** [comment]
On sciserver.org, the home directory has very limited space, and one is supposed to work in workspace/Storage/<username>/persistent/. It would be good to be able to set the astroquery cache directory to another location.

Some workarounds:
  * regularly delete ~/.astropy/cache/download 
  * make ~/.astropy a symlink to another place
  * delete ~/.astropy and create a *file* ~/.astropy, then astroquery cache fails with `WARNING: CacheMissingWarning: Remote data cache could not be accessed due to FileExistsError: [Errno 17] File exists: '/home/idies/.astropy' [astropy.utils.data` and `WARNING: CacheMissingWarning: ("Cache directory cannot be read or created ([Errno 17] File exists: '/home/idies/.astropy'), providing data in temporary file instead.", '/home/idies/workspace/Storage/jbuchner2/chispec/tmp/astropy-download-12547-5hzkwzf3') [astropy.utils.data]` and stores in TMPDIR instead

This is unexpected because I am using `SDSS.get_spectra(..., cache=False)`, so I expected no caching.

A environment variable or other way to influence caching could be useful.

**Comment by keflavich** [comment]
@JohannesBuchner `cache=False` absolutely should result in no files being cached.  When I run `SDSS.query_region(..., cache=False)`, it does not write files - as expected.  I've tested get_spectra, too, and nothing is written to disk:

```python
rslt = SDSS.query_region(SkyCoord(200.5*u.deg, 20*u.deg), radius=45*u.arcsec, cache=False, spectro=True)
spec = SDSS.get_spectra(matches=rslt, cache=False)
```

The other workaround for your situation is to change the `cache_location`:
```python
SDSS.cache_location = 'workspace/Storage//persistent/'
```

If you're seeing a case where `cache=False` is still caching a file, please report that as a separate issue.  As reported right now, I cannot reproduce your problem.

**Comment by bsipocz** [comment]
@ceb8 - is astroquery still on your radar? It would be nice to wrap up the caching issues and make it all consistent and well documented.

**Comment by ceb8** [comment]
@bsipocz Yes. I don't know when I will get back to it. But it's definitely on my radar.
